pallets/identity/src/keys.rs

```diff
@@ -423,6 +431,152 @@ impl<T: Config> Module<T> {
         Ok(())
     }
 
+    /// Create a child identity.
+    pub(crate) fn base_create_child_identity(
+        origin: T::RuntimeOrigin,
+        secondary_key: T::AccountId,
+    ) -> DispatchResult {
+        let (_, parent_did) = Self::ensure_primary_key(origin)?;
+
+        // Make sure `parent_did` has no parent.
+        Self::ensure_no_parent(parent_did)?;
+
+        // Ensure that the key is a secondary key.
+        Self::ensure_secondary_key(parent_did, &secondary_key)?;
+
+        // Ensure that the key can be unlinked.
+        Self::ensure_key_unlinkable_from_did(&secondary_key)?;
+
+        // Generate a new DID for the child.
+        let child_did = Self::make_did()?;
+
+        // Remove links and get all authorization IDs per signer.
+        // Unlink the secondary account key.
+        Self::remove_key_record(&secondary_key, Some(parent_did));
+
+        // All `auth_id`s for `signer` authorized by `parent_did`.
+        let signer = Signatory::Account(secondary_key.clone());
+        for auth_id in Self::auths_of(&signer, parent_did) {
+            // Remove authorizations.
+            Self::unsafe_remove_auth(&signer, auth_id, &parent_did, true);
+        }
```

Comment:
I noticed we don't do this logic when calling `leave_identity_as_key` - wondering whether we should also remove all auths in that case?

Reply:
I understand why the auths should be removed when the primary key removes a secondary key, to make sure there wasn't another "join identity" auth.  Not as important to remove when the key leaves the identity.